sdk/keyvault/Azure.Security.KeyVault.Certificates/src/DeleteCertificateOperation.cs

```diff
@@ -2,35 +2,39 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Certificates
 {
     /// <summary>
     /// A long-running operation for <see cref="CertificateClient.StartDeleteCertificate(string, CancellationToken)"/> or <see cref="CertificateClient.StartDeleteCertificateAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class DeleteCertificateOperation : Operation<DeletedCertificate>
+    public class DeleteCertificateOperation : Operation<DeletedCertificate>, IOperation
     {
         private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(2);
 
         private readonly KeyVaultPipeline _pipeline;
+        private readonly OperationInternal _operationInternal;
         private readonly DeletedCertificate _value;
-        private Response _response;
-        private bool _completed;
 
         internal DeleteCertificateOperation(KeyVaultPipeline pipeline, Response<DeletedCertificate> response)
         {
             _pipeline = pipeline;
             _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
-            _response = response.GetRawResponse();
+            _operationInternal = new(_pipeline.Diagnostics, this, response.GetRawResponse(), nameof(DeleteCertificateOperation), new[]
+            {
+                new KeyValuePair<string, string>("secret", _value.Name), // Retained for backward compatibility.
+                new KeyValuePair<string, string>("certificate", _value.Name),
+            });
 
-            // The recoveryId is only returned if soft-delete is enabled.
+            // The recoveryId is only returned if soft delete is enabled.
             if (_value.RecoveryId is null)
             {
-                _completed = true;
+                // If soft delete is not enabled, deleting is immediate so set success accordingly.
+                _operationInternal.SetState(OperationState.Success(response.GetRawResponse()));
```

Comment:
Why don't you pass the value into SetState here?

Reply:
Because I use `OperationState`, not `OperationState<T>`. I still have to retain my own `_value` I get in the ctor, so there's no value to implement `IOperation<T>`.

Reply:
Can't you use `this.Value` in all those cases?

Reply:
Because if soft delete is enabled I don't go into this code path, but I still need the deserialized value. I can't set value, and I certainly can't set success since I'm still pending. Thse pLROs are different in that `Value` is always available.